cedar-policy-validator/src/schema_file_format.rs

```diff
@@ -1340,31 +1277,31 @@ mod strengthened_types {
             "type": ""
         });
         let schema: Result<ActionEntityUID, _> = serde_json::from_value(src);
-        assert_matches!(schema, Err(err) if &err.to_string() == "invalid action type: unexpected end of input");
+        assert_matches!(schema, Err(err) if &err.to_string() == "invalid name ``: unexpected end of input");
 
         let src = serde_json::json!(
         {
            "id": "action",
             "type": "*"
         });
         let schema: Result<ActionEntityUID, _> = serde_json::from_value(src);
-        assert_matches!(schema, Err(err) if &err.to_string() == "invalid action type: unexpected token `*`");
+        assert_matches!(schema, Err(err) if &err.to_string() == "invalid name `*`: unexpected token `*`");
 
         let src = serde_json::json!(
         {
            "id": "action",
             "type": "Action::"
         });
         let schema: Result<ActionEntityUID, _> = serde_json::from_value(src);
-        assert_matches!(schema, Err(err) if &err.to_string() == "invalid action type: unexpected end of input");
+        assert_matches!(schema, Err(err) if &err.to_string() == "invalid name `Action::`: unexpected end of input");
 
         let src = serde_json::json!(
         {
            "id": "action",
             "type": "::Action"
         });
         let schema: Result<ActionEntityUID, _> = serde_json::from_value(src);
-        assert_matches!(schema, Err(err) if &err.to_string() == "invalid action type: unexpected token `::`");
+        assert_matches!(schema, Err(err) if &err.to_string() == "invalid name `::Action`: unexpected token `::`");
```

Comment:
Would it be easy to make this `invalid action name ...`?

Reply:
while we're here, this assert might be more pretty/readable written without the guard:
```
assert_matches!(schema, Err(err) => {
    assert_eq!(err.to_string, "invalid action name: ...");
});
```

Reply:
> Would it be easy to make this invalid action name ...?

I think we'd need to add back the manual deserialization function for `ActionEntityUID` to get a customized error message here. I don't think it's worth it, but open to other opinions.

Reply:
Seems alright as it is